packages/libs/eda/src/lib/core/hooks/workspace.ts

```diff
@@ -262,3 +269,62 @@ export function useGetDefaultVariableDescriptor() {
     [entities, featuredFields, fieldTree]
   );
 }
+
+export const timeSliderVariableConstraints: DataElementConstraintRecord[] = [
+  {
+    overlayVariable: {
+      isRequired: true,
+      minNumVars: 1,
+      maxNumVars: 1,
+      // TODO: testing with SCORE S. mansoni Cluster Randomized Trial study
+      // however, this study does not have date variable, thus temporarily use below for test purpose
+      // i.e., additionally allowing 'integer'
+      allowedTypes: ['date', 'integer'],
+      // TODO: below two are correct ones
+      // allowedTypes: ['date'],
+      //        isTemporal: true,
+    },
+  },
+];
+
+export function useGetDefaultTimeVariableDescriptor() {
+  const entities = useStudyEntities();
+  // filter constraint for time slider inputVariables component
+
+  return useCallback(
+    function getDefaultTimeVariableDescriptor() {
+      const temporalVariableTree = filterVariablesByConstraint(
+        entities[0],
+        timeSliderVariableConstraints[0]['overlayVariable']
+      );
+
+      // take the first suitable variable from the filtered variable tree
+
+      // first find the first entity with some variables that passed the filter
+      const defaultTimeSliderEntity: StudyEntity | undefined = Array.from(
+        preorder(temporalVariableTree, (entity) => entity.children ?? [])
+      )
+        // not all `variables` are actually variables, so we filter to be sure
+        .filter(
+          (entity) =>
+            entity.variables.filter((variable) => Variable.is(variable))
+              .length > 0
+        )[0];
+
+      // then take the first variable from it
+      const defaultTimeSliderVariable: Variable | undefined =
+        defaultTimeSliderEntity.variables.filter(
+          (variable): variable is Variable => Variable.is(variable)
+        )[0];
```

Comment:
I ran into a runtime bug with this code, when I was playing with constraints. For some reason, typescript is allowing you to assume that `defaultTimeSliderEntity` is defined on line 316, even thought it might be null. The suggested change below will address this, and I think it's also a little more performant (`filter` will iterate the entire array, while `find` and `some` will terminate after the first element for which the callback returns `true`).
```suggestion
      // take the first suitable variable from the filtered variable tree

      // first find the first entity with some variables that passed the filter
      const defaultTimeSliderEntity = Array.from(
        preorder(temporalVariableTree, (node) => node.children ?? [])
      ).find((entity) => entity.variables.some(Variable.is));

      // then take the first variable from it
      const defaultTimeSliderVariable = defaultTimeSliderEntity?.variables.find(
        Variable.is
      );

      return defaultTimeSliderEntity != null &&
        defaultTimeSliderVariable != null
        ? {
            entityId: defaultTimeSliderEntity.id,
            variableId: defaultTimeSliderVariable.id,
          }
        : undefined;
```

Reply:
Furthermore, when this function returns `undefined`, the `MapAnalysisImpl` component enters and infinite loop.